src/quick-lint-js/fe/parse.cpp

```diff
@@ -314,6 +314,51 @@ void parser::error_on_sketchy_condition(expression* ast) {
   }
 }
 
+void parser::error_on_pointless_string_compare(
+    expression::binary_operator* ast) {
+  auto is_comparison_operator = [](string8_view s) {
+    return s == u8"=="sv || s == u8"==="sv || s == u8"!="sv || s == u8"!=="sv;
+  };
+
+  for (int i = 0; i < ast->child_count() - 1; i++) {
+    expression* lhs = ast->child(i);
+    expression* rhs = ast->child(i + 1);
+
+    if ((lhs->kind() == expression_kind::call &&
+         rhs->kind() == expression_kind::literal) ||
+        (lhs->kind() == expression_kind::literal &&
+         rhs->kind() == expression_kind::call)) {
+      source_code_span op_span = ast->operator_spans_[i];
+      if (!is_comparison_operator(op_span.string_view())) {
+        continue;
+      }
+
+      // make sure the call is on the "left" and the literal on the "right"
+      if (lhs->kind() == expression_kind::literal) {
+        expression* tmp = lhs;
+        lhs = rhs;
+        rhs = tmp;
+      }
+
+      string8_view call =
+          lhs->child_0()->variable_identifier().span().string_view();
+      string8_view literal = rhs->span().string_view();
+
+      if (call == u8"toLowerCase"sv) {
+        if (hasupper(literal)) {
```

Comment:
Must fix: `hasupper` is insufficient. A string literal might contain Unicode escapes. Your code will report a warning in the following code, but it shouldn't:
```js
let s = 'M';
console.log(s.toLowerCase() === "\u006D");
```

We currently don't have a way to get the normalized string (i.e. without escapes) from a string literal. So I think we should bail out if we see any `\`.